src/SourceBuild/content/repo-projects/Directory.Build.targets

```diff
@@ -54,6 +40,27 @@
   <UsingTask AssemblyFile="$(XPlatSourceBuildTasksAssembly)" TaskName="WritePackageVersionsProps" />
   <UsingTask AssemblyFile="$(XPlatSourceBuildTasksAssembly)" TaskName="WritePackageUsageData" />
   <UsingTask AssemblyFile="$(XPlatSourceBuildTasksAssembly)" TaskName="WriteUsageReports" />
+  
+  <!-- Wraps the transitive repo references with additional metadata -->
+  <Target Name="GetRepositoryReferenceInfo"
+          DependsOnTargets="GetTransitiveRepositoryReferences">
+    <!-- SBRP is explicitly excluded since it is output to its own special directory, $(ReferencePackagesDir). -->
+    <ItemGroup Condition="'@(TransitiveRepositoryReference)' != ''">
+      <RepositoryReferenceInfo Include="@(TransitiveRepositoryReference)"
+                               Condition="'%(Identity)' != 'source-build-reference-packages'">
+        <ShippingSourceName>source-built-%(Identity)</ShippingSourceName>
+        <NonShippingSourceName>source-built-transport-%(Identity)</NonShippingSourceName>
+        <ShippingPackagesPath>$(ArtifactsShippingPackagesDir)/%(Identity)/</ShippingPackagesPath>
+        <NonShippingPackagesPath>$(ArtifactsNonShippingPackagesDir)/%(Identity)/</NonShippingPackagesPath>
+      </RepositoryReferenceInfo>
+    </ItemGroup>
+    <ItemGroup Condition="'@(RepositoryReferenceInfo)' != ''">
```

Comment:
nit:

```suggestion

    <ItemGroup Condition="'@(RepositoryReferenceInfo)' != ''">
```

Reply:
Fixed in https://github.com/dotnet/installer/pull/18557/commits/241ad6fec829efd38530b3744604fae68f0f5efe